pkg/dao/snapshots.go

```diff
@@ -147,3 +205,61 @@ func (sDao snapshotDaoImpl) FetchLatestSnapshot(repoConfigUUID string) (api.Snap
 	snapshotModelToApi(snap, &apiSnap)
 	return apiSnap, nil
 }
+
+func (sDao *snapshotDaoImpl) getPulpData(ctx context.Context) (string, error) {
+	pulpContentPath, err := sDao.cache.GetPulpContentPath(ctx)
+	if err != nil && !errors.Is(err, cache.NotFound) {
+		log.Logger.Error().Err(err).Msg("Error reading from cache")
+	}
+
+	cacheHit := err == nil
+	if cacheHit {
+		return pulpContentPath, nil
+	}
+
+	if sDao.pulpClient == nil {
+		return "", fmt.Errorf("pulpClient cannot be nil")
+	}
+
+	status, err := sDao.pulpClient.Status()
+	if err != nil {
+		return "", err
+	}
+
+	pulpContentPath = status.ContentSettings.ContentOrigin + status.ContentSettings.ContentPathPrefix
+	err = sDao.cache.SetPulpContentPath(ctx, pulpContentPath)
+	if err != nil {
+		log.Logger.Error().Err(err).Msg("Error writing to cache")
+	}
+
+	return pulpContentPath, nil
+}
+
+func (sDao *snapshotDaoImpl) getContentURL(ctx context.Context, repositoryPath string) (string, error) {
+	pulpContentPath, err := sDao.getPulpData(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	url := pulpContentPath + repositoryPath
```

Comment:
Pulp requires a trailing slash, or else it doesn't really work in pulp (gives a 404).